### PR TITLE
a number of seccomp improvements

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Comment by poettering** [comment]
This adds a number of tests that check if seccomp works correctly on the various archs. For this PR it's hence particularly relevant how the tests pass on the various archs we test.

File: src/test/test-seccomp.c [hunk 1]
```diff
@@ -216,6 +220,259 @@ static void test_restrict_namespace(void) {
...
+
+        if (!is_seccomp_available())
+                return;
+        if (geteuid() != 0)
+                return;
+        /* pid = fork(); */
+        /* assert_se(pid >= 0); */
+
+
+        pid = 0;
```
Comment (phomes) [inline src/test/test-seccomp.c]: Is the fork intentionally commented out?

**Comment by martinpitt** [comment]
Note that our CI currently disables seccomp related unit properties on non-amd64 arches, so to test this properly I [adjusted the hack]( https://anonscm.debian.org/cgit/pkg-systemd/systemd.git/commit/?id=6b9a4937571471) to not apply for this particular PR. Once this lands, I'll backport the commit into our packaging tree and drop the entire hack.

I'll re-run the i386 and s390x tests.

File: src/test/test-seccomp.c [hunk 2]
```diff
@@ -216,6 +220,259 @@ static void test_restrict_namespace(void) {
...
+                assert_se(sched_setscheduler(0, SCHED_RR, &(struct sched_param) { .sched_priority = 1 }) < 0);
+                assert_se(errno == EPERM);
+
+                _exit(EXIT_SUCCESS);
+        }
+
+        assert_se(wait_for_terminate_and_warn("realtimeseccomp", pid, true) == EXIT_SUCCESS);
+}
+
+static void test_memory_dency_write_execute(void) {
```
Comment (martinpitt) [inline src/test/test-seccomp.c]: This is supposed to be "deny"?

File: src/test/test-seccomp.c (hunk 1 above)
Comment (martinpitt) [reply src/test/test-seccomp.c]: Indeed all other tests apply the restrictions into a child process only, and the `wait_for_terminate_and_warn()` call further below is not commented out. This looks like some debugging leftover?

**Comment by martinpitt** [comment]
I verified that the units on non-amd64 now contain `RestrictAddressFamilies=` again (from [artifacts.tar.gz](https://objectstorage.prodstack4-5.canonical.com/v1/AUTH_77e2ada1e7a84929a74ba3b87153c0ac/autopkgtest-xenial-pitti-systemd-semaphore/xenial/s390x/s/systemd-upstream/20161229_104458_a8df5@/artifacts.tar.gz))

**Comment by martinpitt** [comment]
\o/ all green after the retries!

**Comment by mbiebl** [comment]
What's the state of this pull request?
We currently ship systemd v232 in Debian testing where this is broken and the hard freeze is coming soon. So we (as Debian maintainers) need to decide whether to disable that functionality or if we have any upstream fixes which we can backport for v232

**Comment by keszybz** [comment]
I pushed the branch with the suggested fixes.

**Pull request merged**